Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -271,24 +271,24 @@ void SpeculativeJIT::compileFunction()
     setEndOfCode();
 
     // === Link ===
-    auto linkBuffer = makeUnique<LinkBuffer>(*this, m_codeBlock, LinkBuffer::Profile::DFG, JITCompilationCanFail);
-    if (linkBuffer->didFailToAllocate()) {
+    LinkBuffer linkBuffer(*this, m_codeBlock, LinkBuffer::Profile::DFG, JITCompilationCanFail);
+    if (linkBuffer.didFailToAllocate()) {
```

Comment:
Can we mark the LinkBuffer class as `WTF_FORBID_HEAP_ALLOCATION` now and remove its `TZone` data?

Reply:
@kmiller68 Still it is used as a heap allocation in some places, so we cannot use `WTF_FORBID_HEAP_ALLOCATION`.